py/mpconfig.h

```diff
@@ -1434,6 +1434,12 @@ typedef double mp_float_t;
 #define MICROPY_PY_SYS_EXECUTABLE (0)
 #endif
 
+// Whether to provide "sys.identity", implemented simply as returning the argument passed
+// since strings are already interned as qstr.
```

Comment:
I'm guessing this should say "sys.intern".

Also, not all strings are qstrs.  We could actually implement this function by calling the existing `mp_obj_str_intern_checked()`.  That has the benefit of raising an exception if the argument isn't a string (matching CPython).

Reply:
> I'm guessing this should say "sys.intern".

Fixed.

> Also, not all strings are qstrs. We could actually implement this function by calling the existing mp_obj_str_intern_checked()

Yeah was in doubt to use 'most strings are interned' but I missed the existence of that function, changed + added test

Reply:
Right. Must have left my head at home today.